api/mesh/v1alpha1/traffic_route.proto

```diff
@@ -137,15 +137,83 @@ message TrafficRoute {
       map<string, StringMatcher> headers = 3;
     }
 
+    // Modify defines modifications of matched HTTP messages
+    message Modify {
+
+      // RegexReplace defines a way to match string using regex and build a new
+      // one using subsitution section
+      message RegexReplace {
+        // Pattern of the regex using RE2 syntax
+        // https://github.com/google/re2/wiki/Syntax
+        string pattern = 1;
+        // Substitution using regex groups. E.g. use \\1 as a first matched
+        // group
+        string substitution = 2;
+      }
+
+      // Path defines modification of path of the HTTP request
+      message Path {
+        oneof type {
+          // rewritePrefix rewrites previously matched prefix in match section
+          string rewritePrefix = 1;
+          // regex rewrites prefix using regex with substitution
+          RegexReplace regex = 2;
+        }
+      }
+
+      // Host defines modification of the HTTP Host header
+      message Host {
+        oneof type {
+          // value replaces the host header with given value
+          string value = 1;
+          // fromPath replaces the host header from path using regex
+          RegexReplace fromPath = 2;
+        }
+      }
+
+      // Headers defines modification of HTTP headers
+      message Headers {
+        // Add defines operation of adding new HTTP header
+        message Add {
+          // Name of the header
+          string name = 1;
```

Comment:
Should we use generated protobuf generation here? e.g.
```
[(validate.rules).string.well_known_regex = HTTP_HEADER_VALUE]
```

Reply:
Same suggestion for all the new fields :)

Reply:
So the current situation about validators in `.proto` is awkward. We tried using this back in the days and it was very inconvenient - error messages were not user-friendly. Also, it was not that flexible so we ended up writing our own validators. Back when we introduced validators for the first time, the project lead wanted to keep proto validators.
I think we may need to revisit this decision, so we either
1) Remove all validators from proto, because we don't even execute them in the code.
2) Keep the proto validators in proto and try to execute them instead of our validators (at least the one we can)
3) Keep the proto validators in proto and try to execute them after our own validators.